core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -98,6 +98,10 @@ public TcpSlaveAgentListener(int port) throws IOException {
             throw (BindException)new BindException("Failed to listen on port "+port+" because it's already in use.").initCause(e);
         }
         this.configuredPort = port;
+        setUncaughtExceptionHandler((t, e) -> {
+            LOGGER.log(Level.SEVERE, "Uncaught exception in TcpSlaveAgentListener " + t, e);
+            shutdown();
```

Comment:
There is `AperiodicWork` in Jenkins, which can be used to schedule the delayed tasks